source/fundamentals/authentication.txt

```diff
@@ -142,6 +143,180 @@ string as follow:
 
    To learn more on specifying the default mechanism, see :ref:`csharp-authentication-default`.
 
+.. _csharp-aws:
+
+MONGODB-AWS
+~~~~~~~~~~~
+
+.. note::
+
+   The ``MONGODB-AWS`` authentication mechanism is available for
+   MongoDB deployments on MongoDB Atlas.
+
+The ``MONGODB-AWS`` authentication mechanism uses your Amazon Web Services
+Identity and Access Management (AWS IAM) credentials to authenticate your
+user. To learn more about configuring MongoDB Atlas, see the
+:atlas:`Set Up Passwordless Authentication with AWS IAM Roles </security/passwordless-authentication/#set-up-passwordless-authentication-with-aws-iam-roles>` guide.
+
+To instruct the driver to use this authentication mechanism, you can specify
+``MONGODB-AWS`` either as a parameter in the connection string or by using
+the ``MongoCredential`` constructor and passing "MONGODB-AWS" as the authentication mechanism.
+
+Learn how to specify this authentication mechanism and the various ways to
+provide your AWS IAM credentials in the next sections.
+
+These sections contain code examples that use the following placeholders:
+
+* ``awsKeyId`` - value of your AWS access key ID
+* ``awsSecretKey`` - value of your AWS secret access key
+* ``atlasUri`` - hostname and port of your MongoDB Atlas deployment
+* ``awsSessionToken`` - value of your AWS session token
+
+Specify Your Credentials in the Environment
++++++++++++++++++++++++++++++++++++++++++++
+
+You can instruct the driver to automatically retrieve your AWS IAM credentials 
+from the environment by setting the appropriate environment variables.
+
+To use environment variables to supply your credentials, you must perform
+the following steps:
+
+1. Specify the authentication mechanism and authentication source
```

Comment:
Suggestion:
Intuitively, it seems to make sense that authSource is "authentication source", but I don't know whether that's common jargon. The [manual](https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.authSource
) still describes the parameter as a database name (even though "$external" deviates from this).

I think it could be helpful to continue to use "database name" even though the parameter name is "authSource" and the value is "$external" rather than a db name.